components/Table.tsx

```diff
@@ -307,7 +307,8 @@ export const Table: React.FC<Props> = ({ data, hasFilters = true }) => {
       );
       if (newWindow) newWindow.opener = null;
     } else {
-      router.push(`/${row.original.address}`);
+      const chainName = chainIdToNameLookup[row.original.chainId];
+      router.push(`/${chainName}/${row.original.address}`);
```

Comment:
can you handle the case where the chainName might not exist? maybe just default to using the number instead of the name in that case wdyt?